test/connectivity/connectivity.spec.js

```diff
@@ -0,0 +1,305 @@
+  model: {
+    hostname: 'localhost',
+    port: 27020,
+  },
+  expectedInstanceDetails: {
+    _id: 'localhost:27020',
+    authenticatedUsers: [],
+    build: {
+      raw: {
+        version: /4.4.[1-9]+$/
+      }
+    },
+    client: {
+      isWritable: true,
+      isMongos: false,
+    },
+    dataLake: { isDataLake: false, version: null },
+    genuineMongoDB: { isGenuine: true, dbType: 'mongodb' },
+    host: {
+      os: 'Ubuntu',
+      os_family: 'linux',
+      kernel_version: '18.04',
+      arch: 'x86_64'
+    },
+    hostname: 'localhost',
+    port: 27020
+  }
+}];
+
+describe('Connectivity', () => {
+  connectionsToTest.forEach(({
+    dockerComposeFilePath,
+    expectedInstanceDetails,
+    model: testConnectionModel,
+    title: testConnectionTitle
+  }) => {
+    let wrapper;
+    let appRegistry;
+    let compassConnectStore;
+
+    before(async() => {
+      dockerComposeUp(dockerComposeFilePath);
+
+      appRegistry = new AppRegistry();
+
+      activateCompassConnect(appRegistry);
+
+      global.hadronApp = {
+        appRegistry
+      };
+
+      const ROLE = {
+        name: 'Status',
+        component: () => (<div id="statusPlugin">Status</div>)
+      };
+      global.hadronApp.appRegistry = appRegistry;
+      global.hadronApp.appRegistry.registerRole('Application.Status', ROLE);
+
+      compassConnectStore = appRegistry.getStore('Connect.Store');
+
+      // Remove all logic around saving and loading stored connections.
+      // NOTE: This is tightly coupled with the store in compass-connect.
+      // https://github.com/mongodb-js/compass-connect/blob/master/src/stores/index.js
+      compassConnectStore._saveRecent = () => {};
+      compassConnectStore._saveConnection = () => {};
+      compassConnectStore.state.fetchedConnections = [];
+      compassConnectStore.StatusActions = {
+        done: () => {},
+        showIndeterminateProgressBar: () => {}
+      };
+      compassConnectStore.appRegistry = appRegistry;
+
+      // Load the connection into our connection model.
+      const model = new Connection(testConnectionModel);
+      debug('Created model with connection string:', model.driverUrl);
+
+      // Load the connection model through compass-connect and render it.
+      // Load the connection into compass-connect's connection model.
+      compassConnectStore.state.currentConnection = model;
+      compassConnectStore.trigger(compassConnectStore.state);
+
+      // Here we use the parsed connection model and build a url.
+      // This is something that would occur if
+      // a user is switching between the connection views and editing.
+      compassConnectStore.state.customUrl = compassConnectStore.state.currentConnection.driverUrlWithSsh;
+      compassConnectStore.trigger(compassConnectStore.state);
+
+      wrapper = mount(<CompassConnectComponent />);
+
+      // Ensures the model doesn't cause any errors when rendering.
+      await compassConnectStore.validateConnectionString();
```

Comment:
isn't this some kind of `it` as well?

Reply:
Ah yup that can be in the `it` below which is testing the side effects of this method on the store 😅 nice catch